### PR TITLE
Remove context() method from opt::Function

### PR DESCRIPTION
**Pull request by dj2**
This CL removes the context() method from opt::Function. In the places
where the context() was used we can retrieve, or provide, the context in
another fashion.

#1703

### CODE REVIEW

**Comment by dj2** [comment]
Actually, hold off on this until I finish the Pass constructor CL ....

**Comment by dneto0** [comment]
> Actually, hold off on this until I finish the Pass constructor CL ....

Convention is to add WIP to the title.  I did that for this CL.
Remove it when you want a review.

**Comment by s-perron** [comment]
I'm not against this change, but could you create an umbrella issue that ties all of the refactoring changes together with an explanation of what you are trying to accomplish with the changes?  I know we have talked about it a little, but it would be good to make it public.

**Comment by dj2** [comment]
Done. #1703

**Comment by dj2** [comment]
The Pass::Run change has landed so this should be good to go. PTAL.

File: source/opt/dominator_tree.h [hunk 1]
```diff
@@ -158,7 +158,7 @@ class DominatorTree {
 
   // Build the (post-)dominator tree for the function |f|
   // Any existing data will be overwritten
-  void InitializeTree(const opt::Function* f);
+  void InitializeTree(const CFG& cfg, const opt::Function* f);
```
Comment (s-perron) [inline source/opt/dominator_tree.h]: Should add comments about `cfg`.  Make sure to mention that `cfg` includes `f`.

**Pull request merged**